File: flask/app/routes/ums/Role.py
from flask import  Blueprint, jsonify, request
from app.services.ums.RoleService import *
from app.utils.response import *

role_bp = Blueprint('role', __name__, url_prefix='/role')

# 添加
@role_bp.route('/add', methods=['POST'])
def add_role():
    data = request.json
    return create_role(data)

# 编辑
@role_bp.route('/edit/<int:role_id>', methods=['POST'])
def edit_role(role_id):
    data = request.json
    return update_role(role_id, data)

# 删除
@role_bp.route("/delete/<int:role_id>", methods=["POST"])
def delete_role(role_id):
    return remove_role(role_id)


# 模糊分页获取所有
@role_bp.route("/list",methods=["GET"])
def list_role():
    keyword = request.args.get('keyword', ' ')
    page = request.args.get("page", 1, type=int)
    page_size = request.args.get('pageSize', 6, type=int)
    return get_role_list(keyword, page, page_size)

# 启用/禁用
@role_bp.route('/updateStatus/<int:role_id>', methods=['POST'])
def change_status(role_id):
    status = request.json.get("status")
    return update_role_status(role_id, status)

# 获取角色相关菜单
@role_bp.route('/listMenu/<int:role_id>', methods=['GET'])
def role_menu(role_id):
    return get_role_menu(role_id)

# 获取角色相关资源
@role_bp.route('/listResource/<int:role_id>', methods=['GET'])
def role_resource(role_id):
    return get_role_resource(role_id)

# 获取角色权限
@role_bp.route('/listPermission/<int:role_id>', methods=['GET'])
def role_permissions(role_id):
    return get_role_permission(role_id)

# 给角色分配权限（菜单、操作、数据权限）
@role_bp.route('/allocPermission', methods=['POST'])
def allocate_permissions():
    data = request.json
    return allocate_role_permission(data.get("role_id"), data.get("permission_ids", []))

# 给角色分配菜单
@role_bp.route('/allocMenu', methods=['POST'])
def allocate_menu():
    data = request.json
    return allocate_role_menu(data.get("role_id"), data.get("menu_ids",[]))

# 给角色分配资源
@role_bp.route('/allocResource', methods=['POST'])
def allocate_resource():
    data = request.json
    return allocate_role_resource(data.get("role_id"), data.get("resource_ids",[]))

File: vue/src/api/ums/role.js
import request from '@/utils/request'

export function getRoleList(keyword, page, pageSize) {
  return request({
    url: '/role/list',
    method: 'get',
    params: { keyword, page, pageSize }
  })
}

// 添加
export function addRole(data) {
  return request({
    url: '/role/add',
    method: 'post',
    data
  })
}

// 编辑
export function updateRole(role_id, data) {
  return request({
    url: `/role/edit/${role_id}`,
    method: 'post',
    data
  })
}

// 删除
export function deleteRole(role_id) {
  return request({
    url: `/role/delete/${role_id}`,
    method: 'post'
  })
}

// 更新状态
export function updateRoleStatus(role_id, status) {
  return request({
    url: `/role/updateStatus/${role_id}`,
    method: 'post',
    data: { status }
  })
}

// 获取角色菜单
export function getRoleMenu(rolerole_idId) {
  return request({
    url: `/role/listMenu/${role_id}`,
    method: 'get'
  })
}

// 获取角色资源
export function getRoleResource(role_id) {
  return request({
    url: `/role/listResource/${role_id}`,
    method: 'get'
  })
}

// 获取角色权限
export function getRolePermission(role_id) {
  return request({
    url: `/role/listPermission/${role_id}`,
    method: 'get'
  })
}

// 分配角色菜单
export function assignRoleMenu(role_id, menu_ids) {
  return request({
    url: '/role/allocMenu',
    method: 'post',
    data: { role_id: role_id, menu_ids: menu_ids }
  })
}

// 分配角色资源
export function assignRoleResource(role_id, resource_ids) {
  return request({
    url: '/role/allocResource',
    method: 'post',
    data: { role_id: role_id, resource_ids: resource_ids }
  })
}

// 分配角色权限
export function assignRolePermission(role_id, permission_ids) {
  return request({
    url: '/role/allocPermission',
    method: 'post',
    data: { role_id: role_id, permission_ids: permission_ids }
  })
}
